### PR TITLE
Integration with AGCOD

### PR DESCRIPTION
**Pull request by cmgustavo**
* Amazon Gift Code On Demand

### CODE REVIEW

**Comment by HostFat** [comment]
Will it be compatible with other currencies?

It seems that the currencies available are USD, EUR, JPY, CNY and CAD.

File: src/js/services/bwsError.js [hunk 1]
```diff
@@ -139,6 +139,9 @@ angular.module('copayApp.services')
           case 'PASSWORD_INCORRECT':
             body = gettextCatalog.getString('Wrong spending password');
             break;
+          case 'EXCEEDED_DAYLY_LIMIT':
```
Comment (braydonf) [inline src/js/services/bwsError.js]: typo: EXCEEDED_DAILY_LIMIT

File: src/js/services/bwsError.js [hunk 2]
```diff
@@ -139,6 +139,9 @@ angular.module('copayApp.services')
...
+            body = gettextCatalog.getString('Exceeded dayly limit of $500 per user');
```
Comment (braydonf) [inline src/js/services/bwsError.js]: dayly -> daily

File: src/js/services/amazonService.js [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+      if (err) $log.error(err);
+      
+      if (lodash.isEmpty(amazon)) return cb('CAN_NOT_GET_DATA_FROM_STORAGE');
+      
+      if (lodash.isString(amazon)) {
+        amazon = JSON.parse(amazon);
+      }
+
+      if (amazon.date == dateStamp && (amazon.amount + amount) > LIMIT)
+        return cb('EXCEEDED_DAYLY_LIMIT');
```
Comment (braydonf) [inline src/js/services/amazonService.js]: typo: DAILY

Comment (cmgustavo) [inline src/js/services/amazonService.js]: Fixed. Thanks!!

**Comment by matiu** [comment]
Great work!

**Pull request merged**